### PR TITLE
1.17.x edge basic routing suite

### PR DESCRIPTION
**Pull request by sheidkamp**
Backport of https://github.com/solo-io/gloo/pull/9870 and https://github.com/solo-io/gloo/pull/9930

### CODE REVIEW

File: .githooks/pre-commit [hunk 1]
```diff
@@ -2,8 +2,8 @@
 
 echo "Checking commit for JSON web tokens..."
 echo "Note: This fails without python3. Comment out .githooks/pre-commit to disable this hook."
-if test -e /dev/tty ; then
-exec < /dev/tty ./.githooks/pre-commit-python.py
-fi
+# if test -e /dev/tty ; then
```
Comment (jenshu) [inline .githooks/pre-commit:5]: should we just remove the githooks files (i.e. backport https://github.com/solo-io/gloo/pull/9930)?

**Review by jenshu — APPROVED** [review]

**Review by sam-heilbron — APPROVED** [review]

File: .githooks/pre-commit (hunk 1 above)
Comment (sheidkamp) [reply .githooks/pre-commit:5]: Done! (and documented)

Comment (sheidkamp) [reply .githooks/pre-commit:5]: and turns out I hadn't pushed and closing the ticket merged what was there, so will backport separately

**Pull request merged**